maroon/src/epoch.rs

```diff
@@ -0,0 +1,31 @@
+use common::range_key::U64BlobIdClosedInterval;
+use derive_more::Display;
+use sha2::{Digest, Sha256};
+
+#[derive(Debug, Clone, Display)]
+#[display("Epoch {{ increments: {:?}, hash: 0x{:X} }}", increments, hash.iter().fold(0u128, |acc, &x| (acc << 8) | x as u128))]
+pub struct Epoch {
+  pub increments: Vec<U64BlobIdClosedInterval>,
+  pub hash: [u8; 32],
+}
+
+impl Epoch {
+  pub fn new(increments: Vec<U64BlobIdClosedInterval>, prev_hash: Option<[u8; 32]>) -> Epoch {
+    let mut hasher = Sha256::new();
+
+    // Include previous hash if it exists
```

Comment:
This may (should?) be a strict invariant check if I understand your logic correctly: `prev` is `None` if and only if the leftmost index of the leftmost increment is `0`. Right?

Reply:
No. My idea was that previous is None if this one is the first Epoch. Maybe if it's the first epoch - it's better to make a hash from the timestamp?